### ISSUE TITLE
“Module cannot be imported since it has open interaction points” should always point to module

### ISSUE THREAD

**Issue by mietek** [agda/agda#2369]
Agda 2.5.2 likes to say “Module cannot be imported since it has open interaction points”.  This message should include the module name.

It would be nice if the Emacs mode also automatically navigated to the offending module.

**Comment by andreasabel** [comment]
My test case gives an error location, which shows the name of the module.
```agda
module Issue2369.OpenIP where

test : Set
test = {!!}  -- unsolved interaction point
```
```agda
module Issue2369 where

import Issue2369.OpenIP
```
Error:
```
Module cannot be imported since it has open interaction points
when scope checking the declaration
  import Issue2369.OpenIP
```
Could you please upload a MWE which lacks the module name in this error message?

**Comment by mietek** [comment]
Good question. I'm positive I've had a case where there was no message after what I quoted, but I cannot reproduce that right now.

**Comment by andreasabel** [comment]
I think I also encountered this message once without the "when checking" clause.  If you happen to reproduce it, please reopen this issue.  Thanks!

**Comment by mietek** [comment]
The issue seems to be caused by something going wrong inside an `.agdai` file, and so, reproducing it is tricky.

I can provide a set of mostly-empty files that reproduce the issue, however:

- Deleting `Everything.agdai` or causing it to be regenerated will break reproduction;
- Therefore, making any change to `Everything.agda` will break reproduction;
- Therefore, renaming or removing any of the empty module files imported by `Everything.agda` will break reproduction.

The only non-empty files are `Everything.agda`, which contains a number of import statements, and `S4NbE.agda`, which contains an open interaction point.

To reproduce, open `Everything.agda` and hit `C-c C-l`.

https://dl.dropboxusercontent.com/u/95029/Issue2369.zip

**Comment by asr** [comment]
> I can provide a set of mostly-empty files that reproduce the issue 

I couldn't reproduce the issue. After unziping your file I got the error:

```agda
/tmp/Issue2369/Everything.agda:18,8-13
Module cannot be imported since it has open interaction points
(consider adding {-# OPTIONS --allow-unsolved-metas #-} to this
module)
when scope checking the declaration
  import S4NbE
```

**Comment by marco-vassena** [comment]
I also just stumbled upon this issue with Agda version 2.5.3:

>  Module cannot be imported since it has open interaction points
(consider adding {-# OPTIONS --allow-unsolved-metas #-} to this
module)

Let me know if you want me to upload the offending program (it is several files and it would be hard to find a minimal example).

It uses modules parameters, if that could be related to the cause.

**Comment by UlfNorell** [comment]
Does the problem persist if you remove all `agdai` files and try again?

**Comment by marco-vassena** [comment]
No, if I remove all the `agdai` files, then the error reports the problematic module.
In particular I get:

> Module cannot be imported since it has open interaction points
> (consider adding {-# OPTIONS --allow-unsolved-metas #-} to this
> module)
> when scope checking the declaration
>   import FG2CG as .#FG2CG-15572265102019375739 

The name that agda assigns automatically to the module (.#FG2CG-155...) should probably be left out
but it does not hurt.

**Comment by mietek** [comment]
I’ve experienced this issue a number of times since my last comment, but reproduction remains elusive.

**Comment by mietek** [comment]
Still there in Agda 2.5.4.2.  Can we reopen this issue, please?

Here’s the entirety of the message I just received:
```
Module cannot be imported since it has open interaction points
(consider adding {-# OPTIONS --allow-unsolved-metas #-} to this
module)
```

This happened when trying to reload an `Everything` module that imports all other modules.  However, this time, there were no modules with open interaction points in my development at all.  I manually opened and reloaded the other modules, and then reloaded `Everything`, which worked without complaint.

Therefore, I suspect there is a problem in the code that decides whether an `.agdai` file is stale or not.

**Comment by gallais** [comment]
We probably still need a MWE to be able to make progress on this front.

**Comment by mietek** [comment]
<img width="656" alt="Screen Shot 2021-04-03 at 07 21 52" src="https://user-images.githubusercontent.com/166207/113469375-4a027600-944d-11eb-8397-51644589a54a.png">

Still there in 2.6.1.3.

**Comment by gallais** [comment]
Is it reproducible? If so can you please test https://github.com/gallais/agda/commit/ad5a8c2cad00d8d6fcf67520030cb15dfd5ae32b ?

I wonder whether we can do away with this error completely and simply raise a warning.
It may be necessary to check `--allow-unsolved-metas` is not incompatible with the file's
options though otherwise we may end up accepting holey files as if they were `--safe`...

**Comment by andreasabel** [comment]
Fixed in 2.9.0 by
- #8135

**Comment by mietek** [comment]
Amazing! Thank you.
